src/utils/experimentUpdatesHandler.js

```diff
@@ -42,7 +42,7 @@ const experimentUpdatesHandler = (dispatch) => (experimentId, update) => {
 const onQCUpdate = (update, dispatch, experimentId) => {
   const { input, output } = update;
 
-  const processingConfigUpdate = output.config;
+  const processingConfigUpdate = output?.config;
```

Comment:
when is the `output` null? is it a valid state or an error we should fix instead of using the conditional chaining?

Reply:
The output is `null` when moving from GEM2S to QC in `staging` or `production` due to a call to `assignPodToPipeline` returning a message to the UI. This can not be reproduced in local because we emulate step functions using lambdas instead of step functions. I had a discussion about this with Iva about a month ago. A screenshot of the message can be seen [here](https://files.slack.com/files-pri/T0155DZFVM0-F033B39AWLX/image.png).

I can open a ticket to fix this.

Reply:
I think it's ok to fix it like this. When we decided to notify (or rather not filter out) the `assignPodToPipeline` call, the reasoning was that maybe it would be useful for the UI to know that a pod was already assigned and display it to the user as another step or something like that. I don't think we will do that in the short-term but I don't think it's necessary to add extra code to filter them out. This looks like the best solution